stackle_api/app/models/post.js

```diff
@@ -298,6 +274,89 @@ postSchema.statics.removeTag = function(request, response){
         }
 }
 
+//POST- like the post
+postSchema.statics.setLikeUp = function(request, response){
+   
+    try{
+        //Valdiating postId in req.params
+        const validatorPostId = new Validator(request.params);
+        const input1 = validatorPostId.validatePostId();
+        const currentPostId = input1.postId;
+        
+        //Valdiating userId in req.body   
+        const validatorUserId = new Validator(request.body);
+        const input2 = validatorUserId.validateUserId();
+        const currentUserId = input2.userId;
+
+        this.findOne({_id: currentPostId}, (error, result) => {
+                if (error) {
+                    return returnWithResponse.configureReturnData({ status: 400, success: false, result: error }, response);
+                }
+
+                if(!result)
+                   return returnWithResponse.configureReturnData({ status: 400, success: false, result: `Post-${currentPostId} not found`}, response);
+
+               var likeArray = result.likes;
+               var check = likeArray.indexOf(currentUserId);
+              
+               if(check === -1){
+                    result.likes.push(currentUserId);
+                    result.save();
+                    return returnWithResponse.configureReturnData({ status: 200, success: true, result: `${currentUserId}: user liked the post-${currentPostId}`}, response);
```

Comment:
@psnmissaka this would solve the double liking issue

Reply:
@ntbandara3 update the front-end accordingly. If someone has liked a post and presses the like button again then trigger the unlike.

Reply:
@psnmissaka sure. working on it.